src/renderer/components/+preferences/extensions.tsx

```diff
@@ -36,8 +46,16 @@ export const Extensions = withInjectables<Dependencies>(
   observer(NonInjectedExtensions),
 
   {
-    getProps: (di) => ({
-      preferenceItems: di.inject(extensionsPreferenceItemsInjectable),
-    }),
+    getProps: (di) => {
+      const pathParameters = di.inject(currentPathParametersInjectable);
+      const extensionId = pathParameters.get().extensionId;
+      const extensions = di.inject(rendererExtensionsInjectable);
+      const extension = extensions.get().find((extension) => extension.sanitizedExtensionId === extensionId);
```

Comment:
Because these won't be done within an observer component I don't think this will be reacted to. Therefore it should be an injectable of its own (resolving to some computed).

Reply:
Indeed dereferencing is in wrong place. All deferencing should happen inside reactive context (e.g. in observer). 

Here it could be:


**extensionPreferencesModelInjectable**
```
const extensionPreferencesModel = getInjectable({
  id: 'extension-preferences-model',

  instantiate: (di) => {
    const pathParameters = di.inject(currentPathParametersInjectable);
    const extensions = di.inject(rendererExtensionsInjectable);
      
    return computed(() => {
      const extensionId = pathParameters.get().extensionId;
      const targetExtension = extensions.get().find((extension) => extension.sanitizedExtensionId === extensionId);
      
      if (!targetExtension) {
        throw new Error('Tried to get extension preferences for extension that does not exist.')
      }
      
      return {
        extensionName: targetExtension.manifest.name,
        preferenceItems: getExtensionPreferenceItems(targetExtension)
      }
    });
  }
});
```


Then `getProps` in component could be. 

```
return {
  model: di.inject(extensionPreferencesModelInjectable)
};
```

In that way, we have one dependency for the page. Which is page model which composed stuff from other places. And rereferencing reactive values happen inside reactive context.